### PR TITLE
discoverd: Only log failed HTTP requests

### PR DESCRIPTION
**Pull request by josephglanville**
When logging all successful heartbeats etc the discoverd log can become overly verbose.
This change set removes logging of successful HTTP requests and redirects in favour of only logging failures.
We may need to add explicit logging for certain events to compensate for this if we find it necessary.

### CODE REVIEW

File: discoverd/server/handler.go [hunk 1]
```diff
@@ -22,6 +23,15 @@ import (
 // StreamBufferSize is the size of the channel buffer used for event subscription.
 const StreamBufferSize = 64 // TODO: Figure out how big this buffer should be.
 
+func loggerFn(handler http.Handler, logger log.Logger, clientIP string, rw *hh.ResponseWriter, req *http.Request) {
+	handler.ServeHTTP(rw, req)
+	switch rw.Status() {
+	case 200, 307, 0: // If the status code is 0 it means no response header was written and http.Server will set StatusOK
+	default:
+		logger.Info("request failed", "status", rw.Status())
```
Comment (titanous) [inline discoverd/server/handler.go]: This should log all of the same fields that the default logger does.

**Comment by josephglanville** [comment]
@titanous corrected this so it matches normal logs, minus the fact that the starting request log message is omitted.

File: discoverd/server/handler.go [hunk 2]
```diff
@@ -22,6 +24,16 @@ import (
 // StreamBufferSize is the size of the channel buffer used for event subscription.
 const StreamBufferSize = 64 // TODO: Figure out how big this buffer should be.
 
+func loggerFn(handler http.Handler, logger log.Logger, clientIP string, rw *hh.ResponseWriter, req *http.Request) {
+	start := time.Now()
+	handler.ServeHTTP(rw, req)
+	switch rw.Status() {
+	case 200, 307, 0: // 0 == 200 OK
+	default:
+		logger.Info("request completed", "status", rw.Status(), "duration", time.Since(start))
```
Comment (titanous) [inline discoverd/server/handler.go]: Since this doesn't log the request start, it should also log th request method/path/client ip.

Comment (josephglanville) [inline discoverd/server/handler.go]: Fixed.

**Comment by titanous** [comment]
LGTM

**Pull request merged**